distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso

```diff
@@ -1947,9 +2063,9 @@ type DB_Column
        name. If the column is not floating point, just return the expression.
     short_circuit_special_floating_point : DB_Column -> DB_Column
     short_circuit_special_floating_point self exp =
-        self_is_nan = if self.connection.dialect.supports_separate_nan then self.is_nan else self.is_nothing
         if self.value_type.is_floating_point.not then exp else
-            ((self_is_nan || self.is_infinite).iif self exp).rename exp.name
+            (self.is_finite.not.iif self exp).rename exp.name
```

Comment:
I originally also used `with` on `iif`, since `iif` duplicates its boolean argument. I took this out, because it was applying the CTE transform to expressions that did not need it, thus making queries larger and more complicated. (As noted in the `with` documentation, `with` should be used in downstream logic such as `round`, rather than inside core library methods like `iif`.)

I kept it in `short_circuit_special_floating_point` because it is used in many library methods and generates significant overhead.

Reply:
> `with` should be used in downstream logic such as `round`, rather than inside core library methods like `iif`

I don't think I understand - what is the difference here between `round` and `iif`? Both are regular library methods.

Reply:
The difference is that `iif` is used in many other methods, while `round` is usually called directly by the user. If `iif` always used a CTE, then it would often be used on very small inputs, which would result in a larger query, and there would be no easy way to turn it off. (It could definitely be an optional parameter to `iif` or any other method.)

Also, `round` is large and complicated, and even for small inputs, it generates many duplicates, so it will always be useful to use `with` in several places within it. (But not all places -- I tried many combinations, out of curiosity, and some of them made it worse so I removed those.) So that's what I mean when I say `round` is "downstream" from core library code -- it's larger, more specific, and is not used in many methods throughout the library, unlike `iif`.

I guess the rule is always "use CTEs if it makes the query smaller". For a method like `iif`, there is no way for `iif` to know whether it will always help, but for `round` I think it always will.

We could also examine the "size" of the input to determine if it's necessary -- I think that eventually we will probably want to do that. But for now, the best approach is to experiment by adding `with` calls and checking the result. A more by-hand process, but acceptable for complex methods.

Reply:
Ok, that sounds good.